scripts/gitlab/build_and_test.sh

```diff
@@ -102,10 +102,19 @@ then
     rm -rf ${build_dir} 2>/dev/null
     mkdir -p ${build_dir} && cd ${build_dir}
 
-    cmake \
-      -C ${hostconfig_path} \
-      ${project_dir}
-    cmake --build . -j 32
+    if [[ ${sys_type} == *toss_3* ]]; then
+        cmake \
+        -DENABLE_CCACHE=ON \
+        -DCCACHE_DIR=/collab/usr/global/tools/ccache/toss_3_x86_64_ib/ccache-4.2/bin/ccache \
```

Comment:
Does ccache work well with a single location that may be targeted by several builds simultaneously? (Maybe it creates a unique subdir per configuration hash?)
Also, in the log I see:
```
CMake Warning:
  Manually-specified variables were not used by the project:
    CCACHE_DIR
```
Are we sure this is taken into account by BLT?

Reply:
Oops, that's the binary location, not the location of the data...

Then, where is the data stored? If it's in the working directory, then I'm afraid this won't help much. That's because the working directory changes with the runner hash, the job rank on the runner (0 to 7), and then we have no guarantee that the same target will be rebuilt there.

This is not a lost cause though.

Reply:
That's a typo in my BLT changes :D and we can change the location via a config file. I am trying to work out the best way to set up the config, I think we will need to it via env var.